setup.py

```diff
@@ -6,6 +6,36 @@
 
 NOTE: Put all metadata in pyproject.toml.
 """
+import os
+import pathlib
 import setuptools
+import sys
 
-setuptools.setup()
+from collections import namedtuple
+from datetime import date
+
+TOP_DIR = os.path.realpath(os.path.dirname(__file__))
+PACKAGE_NAME = "onnx_script"
+
+with open(os.path.join(TOP_DIR, "VERSION_NUMBER")) as version_file:
+    VERSION_NUMBER = version_file.read().strip()
+    if "--weekly_build" in sys.argv:
+        today_number = date.today().strftime("%Y%m%d")
+        VERSION_NUMBER += ".dev" + today_number
+        PACKAGE_NAME = "onnx_script_weekly"
+        sys.argv.remove("--weekly_build")
+
+packages = setuptools.find_packages() + setuptools.find_namespace_packages(
+    include="onnxscript"
+)
+
```

Comment:
```suggestion
```

Reply:
nit: This is done by pyproject.toml. Anything that doesn't need to happen in setup.py shouldn't happen in it imo.